src/pages/Settings/UsersAndPermissions/Groups/RoleSelectField.tsx

```diff
@@ -0,0 +1,135 @@
+import Paper from "@mui/material/Paper";
+import Box from "@mui/material/Box";
+import FormLabel from "@mui/material/FormLabel";
+import Stack from "@mui/material/Stack";
+import { startCase } from "lodash-es";
+import FormControlLabel from "@mui/material/FormControlLabel";
+import Checkbox from "@mui/material/Checkbox";
+import Accordion from "@mui/material/Accordion";
+import AccordionSummary from "@mui/material/AccordionSummary";
+import AccordionDetails from "@mui/material/AccordionDetails";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import Typography from "@mui/material/Typography";
+import { ChangeEvent, useState } from "react";
+import { FieldProps } from "formik";
+
+import { useShop } from "@containers/ShopProvider";
+import { useGetRolesQuery } from "@graphql/generates";
+import { client } from "@graphql/graphql-request-client";
+import { InputWithLabel } from "@components/TextField";
+import { filterNodes } from "@utils/common";
+
+type RoleItem = {
+  name: string
+  label: string
+}
+
+const normalizeRoles = (roleNames: string[]) => {
+  const groupedRoleByResource = roleNames.reduce((roleByResource, roleName) => {
+    const [resource, action] = roleName.split("/");
+    (roleByResource[resource] || (roleByResource[resource] = [])).push({ name: roleName, label: startCase(action) });
+    return roleByResource;
+  }, {} as Record<string, RoleItem[]>);
+
+  return groupedRoleByResource;
+};
+
+const getResourceLabel = (name: string) => startCase(name.split(":")[2] || "Unknown");
+
+type RoleSelectFieldProps = FieldProps<Record<string, string[]>> & {
+  predefinedRoles: string[]
+}
+
+export const RoleSelectField = ({ field: { value: selectedRoles, name: fieldName }, form: { setFieldValue }, predefinedRoles }: RoleSelectFieldProps) => {
+  const { shopId } = useShop();
+
+  const [allRolesByResource, setRolesByResource] =
+  useState(normalizeRoles(predefinedRoles));
+
+
+  const { data } = useGetRolesQuery(client, { shopId: shopId! }, {
```

Comment:
Currently, there are 88 roles. I think for the sake of simplicity we can hard code to fetch the first 200 roles because I'm pretty positive the number of roles won't exceed 200 even with a lot of custom plugins that define their roles. An alternative is to create a function that makes query requests until there are no roles unfetched but this is overkill for now. Ideally, more data should be fetched the more you scroll but this will be way too much effort for the Roles entity and the first approach is e good balance between functionality and simplicity.
```suggestion
  const { data } = useGetRolesQuery(client, { shopId: shopId!, first: 200 }, {
```

Reply:
Makes sense, I thought that if we don't pass the first then it will return all records. That's a wrong assumption though.